storage/aptosdb/src/lib.rs

```diff
@@ -167,23 +171,18 @@ fn error_if_too_many_requested(num_requested: u64, max_allowed: u64) -> Result<(
 }
 
 fn error_if_version_is_pruned(
-    pruner: &Option<Pruner>,
-    pruner_index: PrunerIndex,
+    pruner: &(dyn PrunerManager + Send + Sync),
     data_type: &str,
     version: Version,
 ) -> Result<()> {
-    if let Some(pruner) = pruner.as_ref() {
-        if let Some(min_readable_version) =
-            pruner.get_min_readable_version_by_pruner_index(pruner_index)
-        {
-            ensure!(
-                version >= min_readable_version,
-                "{} version {} is pruned, min available version is {}.",
-                data_type,
-                version,
-                min_readable_version
-            );
-        }
+    if let Some(min_readable_version) = pruner.get_min_readable_version() {
```

Comment:
I'm a bit confused what it means when min_readable_version is None... could you remind me about it? I feel it's ideal if we define that precisely.

Reply:
I think it is none when the pruner is disabled.

Reply:
Not the case any more, right? You either don't have `pruner` to call the method on at all if you keep the `Option<XPrunerManager>` approach, or have the min_readable_version initialized in the `manager always there with optional worker` approach.

Reply:
Will follow up PR to address this.